backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -805,6 +821,96 @@ private CommitEntity getCommitEntityObj(
     }
   }
 
+  @Override
+  public VersioningLineageEntry getVersioningEntryByBlob(
+      Session session,
+      RepositoryFunction repositoryFunction,
+      VersioningBlobElement versioningBlobElement)
+      throws ModelDBException {
+    RepositoryEntity repositoryEntity = repositoryFunction.apply(session);
+    String blobSha = versioningBlobElement.getBlobSha();
+    String blobType = versioningBlobElement.getBlobType();
+    List list = getInternalFolderElementEntity(session, blobSha, blobType);
+    for (Object entity : list) {
+      InternalFolderElementEntity internalFolderElementEntity =
+          (InternalFolderElementEntity) entity;
+      VersioningLineageEntry result =
+          getCommit(
+              session,
+              internalFolderElementEntity,
+              VersioningLineageEntry.newBuilder()
+                  .addLocation(internalFolderElementEntity.getElement_name()),
+              repositoryEntity);
+      if (result != null) {
+        return result;
+      }
+    }
+    throw new ModelDBException("Commit not found", Status.Code.NOT_FOUND);
+  }
+
+  private VersioningLineageEntry getCommit(
```

Comment:
I got lost in this method even after trying to follow a few times. I'm also not familiar with the code it's calling. Can you add comments on what it's doing?

Reply:
added